File: com-mini-spring-core/src/main/java/com/mini/jdbc/Paging.java
package com.mini.jdbc;

public final class Paging {
    private int page;
    private int rows;
    private int total;

    public Paging() {}

    public Paging(int page, int rows) {
        this.page = page;
        this.rows = rows;
    }

    public int getPage() {
        return Math.max(1, page);
    }

    public Paging setPage(int page) {
        this.page = page;
        return this;
    }

    public int getRows() {
        return Math.max(1, rows);
    }

    public Paging setRows(int rows) {
        this.rows = rows;
        return this;
    }

    public int getTotal() {
        return Math.max(0, total);
    }

    public Paging setTotal(int total) {
        this.total = total;
        return this;
    }

    public int getStart() {
        return (page - 1) * rows;
    }

    public int getMaxPage() {
        return (int) Math.ceil(total * 1.0 / rows);
    }

}

File: sn.mini.kotlin.util/src/main/java/sn/mini/kotlin/util/Paging.kt
package sn.mini.kotlin.util


import kotlin.math.ceil
import kotlin.math.max
import kotlin.math.min


fun main(args: Array<String>) {
    val ag = arrayOf(1, 2, 3)

    ag.let {
        if (it.isNotEmpty()) {
            println(it[0])
            return@let Unit
        }
        println(it.size)
    }
//
//    ag.forEach {
//        if (it == 2) {
//            println(it)
//            return@forEach Unit
//        }
//        println(it)
//    }
}

data class Paging(private val mPage: Int, private val mRows: Int) {

    var page: Int = mPage
        get() = max(1, min(field, maxPage))

    var rows: Int = mRows
        get() = max(1, field)

    var total: Int = 0
        get() = max(0, field)

    val start = (page - 1) * rows

    val maxPage = ceil(this.total * 1.0 / this.rows).toInt()
}

